schema/metric.go

```diff
@@ -54,6 +56,9 @@ func (m *MetricData) Validate() error {
 	if m.Mtype == "" || (m.Mtype != "gauge" && m.Mtype != "rate" && m.Mtype != "count" && m.Mtype != "counter" && m.Mtype != "timestamp") {
 		return ErrInvalidMtype
 	}
+	if !utf8.ValidString(m.Name) {
```

Comment:
if a name is not valid utf8 then this would now return without validating the tags. if invalid utf8 rejection is disabled, then we still need to validate the tags. 

also, the tags should also be checked for invalid utf8, not only the name.

Reply:
Yeah, I'll rework this a bit